mcs/class/Mono.Security/Mono.Security.X509/PKCS12.cs

```diff
@@ -383,8 +383,13 @@ private void Decode (byte[] data)
 
 				byte[] authSafeData = authSafe.Content [0].Value;
 				byte[] calculatedMac = MAC (_password, macSalt.Value, _iterations, authSafeData);
-				if (!Compare (macValue, calculatedMac))
-					throw new CryptographicException ("Invalid MAC - file may have been tampered!");
+				if (!Compare (macValue, calculatedMac)) {
```

Comment:
Thanks for sorting out my screw-up. I+IBk-ll remember to branch next time.

Neale

From:  Alexander K+APY-plinger <notifications@github.com>
Reply-To:  mono/mono
<reply++0004b6fad2defbf9ba4bc46c3a42d09198f539b4922fab1d92cf0000000112a7978f9
2a170ce02eeb770@reply.github.com>
Date:  Friday, January 8, 2016 at 10:28 AM
To:  mono/mono <mono@noreply.github.com>
Cc:  Neale Ferguson <nealeferguson@verizon.net>
Subject:  Re: [mono] Implement a SetPriority/GetPriority facility for
real-time threads (#2378)


In mcs/class/Mono.Security/Mono.Security.X509/PKCS12.cs
<https://github.com/mono/mono/pull/2378#discussion_r49198960> :
> @@ -383,8 ++383,13 @@ private void Decode (byte[] data)
>  
>  				byte[] authSafeData = authSafe.Content [0].Value;
>  				byte[] calculatedMac = MAC (_password, macSalt.Value, _iterations,
authSafeData);
> -				if (!Compare (macValue, calculatedMac))
> -					throw new CryptographicException ("Invalid MAC - file may have been
tampered!");
> ++				if (!Compare (macValue, calculatedMac)) {
I don't think they should be in here.

@nealef <https://github.com/nealef>  to avoid this in the future you should
push your changes to a new branch and open the PR from there, instead of
working on master (GitHub will update a PR when new commits are pushed to a
branch).

@tritao <https://github.com/tritao>  can you please cherry-pick the commit?

+IBQ
Reply to this email directly or view it on GitHub
<https://github.com/mono/mono/pull/2378/files#r49198960> .